import numpy as np
import pandas as pd
import sys
sys.path.append('..')
from models.feature_extraction.feature_extraction import Data

In [2]:
data = Data()
data.LL_data = pd.read_csv('../LL_data.csv', index_col=0)
data.SL_data = pd.read_csv('../SL_data.csv', index_col=0)
data.matches = pd.read_csv('../matches.csv', parse_dates=['Date','Open'], index_col=0)
data.match_data = pd.read_csv('../match_data.csv', parse_dates=['Date'],index_col=0)

`home_advantage` is $\frac{P(\text{win}\mid \text{home})}{P(\text{win})}$. That is the improvement of the team when playing home. The correlation represents the influence of the teams better play in the home matches than average win rate overall on its winning in a home match.

In [4]:
home_advantage = []
for match,HID in zip(data.matches.index,data.matches.HID):
    home_advantage.append(data.home_advantage(HID,MatchID=match))

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches[

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extrac

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_w

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:542: RuntimeWarning: invalid value encountered in double_scalars
  return home_win_r / win_r # test only away_lose_r
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Dr

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

In [8]:
home_advantage = pd.Series(home_advantage)
home_advantage.corr(data.matches.H + data.matches.D * 0.5)

-0.006206082416233531

$-0.006206082416233531$ is the opposite correlation than I was expecting and in any case, it is pretty insignificant.

`home_win_rate` is $P(\text{win} \mid \text{home})$. The correlation represents the significance of the previous home matches on the result in another home match.

In [12]:
home_win_rate = []
for match,HID in zip(data.matches.index,data.matches.HID):
    home_win_rate.append(data.home_advantage(HID,MatchID=match,rate=True))

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: inva

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:537: 

../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountered in double_scalars
  home_win_r = (previous_home_matches['M_Win'] + previous_home_matches['M_Draw'] * .5).sum()/len(previous_home_matches)
../models/feature_extraction/feature_extraction.py:538: RuntimeWarning: invalid value encountered in double_scalars
  win_r = (previous_matches['M_Win'] + previous_matches['M_Draw'] * .5).sum()/len(previous_matches)
../models/feature_extraction/feature_extraction.py:537: RuntimeWarning: invalid value encountere

In [14]:
home_win_rate = pd.Series(home_win_rate)
home_win_rate.corr(data.matches.H + data.matches.D * 0.5)

0.01585879513442969

$0.01585879513442969$ is not as a significant correlation as would be ideal.